lib/rubocop/ast/node/instance_variable_node.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # A node extension for `sym` nodes. This will be used in  place of a
+    # plain node when the builder constructs the AST, making its methods
+    # available to all `sym` nodes within RuboCop.
+    class InstanceVariableNode < Node
+      include BasicLiteralNode
+
+      # @return String the name of the instance variable
+      def name
+        children.first.to_s.sub('@', '')
```

Comment:
I'm not sure it should return a "String" (not "Symbol"), and it should trim`@`.
I think "Instance variable name" is like `:@foo`. For example

```
irb(main):001:0> class C ; def initialize; @foo = 1 end end 
=> :initialize
irb(main):002:0> C.new.instance_variables
=> [:@foo]
```

So, I guess we will be confused by the returned value of `InstanceVariableNode#name` method.

Reply:
I converted to string and remove the `@` because I had to do it in my [other PR](https://github.com/bbatsov/rubocop/pull/5445)

I agree with you, it's confusing. Would it be ok if `name` returned `:@foo` and there was another method, not sure what to call it, that returned `foo`? Method name suggestions welcome!

Reply:
Sounds good @satyap. The danger of introducing abstractions as part of other changes is just that. The abstraction easily gets coupled to a single use case by accident. 🙂

Reply:
Also, instead of using `children.first`, generally there will be a method `#node_parts` that destructures using `*`, and the other methods make use of that. See the other node extensions for examples.

Example:

```ruby
def identifier
  node_parts[0]
end

def name
  identifier.to_s[1..-1]
end

def node_parts
  to_a
end
```

(I'm not sure I'm recommending these method names. It's just an example.)